cookbooks/bcpc/recipes/keepalived.rb

```diff
@@ -20,7 +20,23 @@
 include_recipe "bcpc::default"
 
 make_config('keepalived-router-id', "#{(rand * 1000).to_i%254/2*2+1}")
-make_config('keepalived-password', secure_password)
+
+keepalived_password = get_config('keepalived-password')
+if keepalived_password.nil?
+  keepalived_password = secure_password
+end
+
+bootstrap = get_all_nodes.select{|s| s.hostname.include? 'bootstrap'}[0].fqdn
```

Comment:
For performance, instead of doing a `get_all_nodes` for the bootstrap 7+ times (once each recipe); perhaps create a cache? Either a memoized helper function or an "ephemeral" node attribute?